src/database/DataTypes/StocktakeBatch.js

```diff
@@ -211,7 +211,25 @@ export class StocktakeBatch extends Realm.Object {
    */
   set countedTotalQuantity(quantity) {
     // Handle packsize of 0.
-    this.countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+    const countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+
+    this.countedNumberOfPacks = countedNumberOfPacks;
+    this.doses = countedNumberOfPacks * this.dosesPerVial;
+  }
+
+  get isVaccine() {
+    return this.itemBatch?.item?.isVaccine ?? false;
+  }
+
+  get dosesPerVial() {
+    return this.isVaccine ? this.itemBatch?.item?.doses ?? 0 : 0;
+  }
+
+  setDoses(database, newValue) {
+    const maximumDosesPossible = this.dosesPerVial * this.countedTotalQuantity;
+    this.doses = Math.min(newValue, maximumDosesPossible);
+
+    database.save('StocktakeBatch', this);
```

Comment:
Should the saving be done by the caller? Then could just make this a tidy setter?

Reply:
I'm struggling - the negative side effects of not using `database.save` is pretty big but I know the setter is nicer. I don't knowwww